deposit/hal/metadata.py

```diff
@@ -148,7 +148,10 @@ def render(self, paper, filename, form):
         language.attrib['ident'] = 'en'  # TODO adapt this?
         textClass = addChild(profileDesc, 'textClass')
 
-        domains = [form.cleaned_data['topic'].lower()]
+        domains = []
+        if 'cleaned_data' in dir(form):
```

Comment:
`MetadataFormatter.toString()` has `form` as an optional argument, but in practice it is currently not omitted. Furthermore, HAL requires a topic/domain to be given for every paper, which we obtain from that form. To fully cover HALs specification, tests should provide a (dummy) topic. 

Rather than a form, `MetadataFormatter`'s methods could take the `cleaned_data` dictionary, which is simpler to mock, perhaps?